src/mcu.cpp

```diff
@@ -1180,9 +1191,25 @@ int main(int argc, char *argv[])
             {
                 port = atoi(argv[i] + 3);
             }
-            else if (!strncmp(argv[i], "-a:", 3))
+            else if (!strncmp(argv[i], "-d:", 3))
+            {
+                char* pColon = argv[i];
+                
+                do {
+                    if (!(pColon = strchr(pColon, ':')) || pColon[1] == 0) break;
+                    audioDeviceIndex = atoi(++pColon);
+                } while (0);
+            }
+            else if (!strncmp(argv[i], "-ab:", 4))
             {
-                audioDeviceIndex = atoi(argv[i] + 3);
+                char* pColon = argv[i];
+                
+                do {
+                    if (!(pColon = strchr(pColon, ':')) || pColon[1] == 0) break;
```

Comment:
first strchr is redundant because we know that it is exactly on argv[i]+3

Reply:
The do + strchr + break thing prevents passing things like -a:or -a.
Very clever! But maybe too clever when debugging later?

Reply:
Let me know how you feel about the remaining shenanigans.